### DISCUSSION TITLE
Dynamic routes with slashes - does not work

### DISCUSSION THREAD

**Discussion by mattbadley** [vercel/next.js#54850] [Help]
### Summary

I have the following folder structure:

```/pages/[category]/guides/[slug].tsx```

When [category] is a string with no slashes, then the page is correctly created, e.g:

```http://localhost:3000/shoes/guides/how-to-wear-a-shoe ```
```http://localhost:3000/hats/guides/how-to-wear-a-hat```

But if [category] contains slashes, then the build fails:

```http://localhost:3000/shoes/sandals/guides/how-to-wear-a-sandal```

Is this a bug, or intentional, or a feature that should be added?

If this structure is not possible using dynamic routes, then how can I achieve the structure above?

**Comment by devjiwonchoi** [comment]
> `http://localhost:3000/shoes/sandals/guides/how-to-wear-a-sandal`

The dynamic route folder `[category]` catches only until `shoes`, and when `sandals` comes in, it is invalid since the next route should be `guides`.

This is not a recommended way to set route segments, and also assuming you can't use catch-all segments like `[...category]`.

I'd recommend you move the guides page outside.

`/guides/shoes/sandals/how-to-wear-a-sandal` as `/pages/guides/[...category].tsx`

**Reply by mattbadley** [reply]
Thanks for the response - your solution might need to be what is implemented, but obviously means changing page URLs to move the 'guide' part elsewhere in the URL.

If it's essential to keep the URL structure in tact, then what options do I have?  

Also seems strange that you can only have catch all segments at the end of a path, is there a reason for that?

**Reply by devjiwonchoi** [reply]
> Also seems strange that you can only have catch all segments at the end of a path, is there a reason for that?

Catch all literally means to "catch all routes starting from...".

For example, `/guides/[...slug]` catches all routes under `/guides` as `/guides/a/b/c/d/e/f/g/...`

So if you have something like `/guides/[...slug]/some-dir/index.tsx` this will cause an error.

### Alternate, a bit annoying solution

A quick way to resolve it is to have `guides` on both `category` and `product`.

`/pages/[category]/guides/[slug].tsx` and `/pages/[category]/[product]/guides/[slug].tsx`
